### PR TITLE
brew-report-issue: cleanup `git credential` usage.

### PR DESCRIPTION
**Pull request by mikemcquaid**
@peff pointed out to me that you can use `git credential approve` to simplify some of this code.

CC @peff to see if this is what you had in mind.

Closes #4.

### CODE REVIEW

**Comment by peff** [comment]
Yeah, this mirrors how git itself uses the credential helpers.

File: cmd/brew-report-issue.rb [hunk 1]
```diff
@@ -13,19 +13,21 @@
...
+if `git config --global credential.helper`.chomp.empty?
   abort <<-EOS
 Error: your Git HTTP(S) credential helper is not set! Set it by running Strap:
 #{strap_url}
 EOS
 end
 
-@github_credentials=`printf "protocol=https\nhost=github.com\n" | git credential fill 2>/dev/null`
+def credential_helper(command, input)
+  `printf "#{input}" | git credential #{command} 2>/dev/null`
```
Comment (peff) [inline cmd/brew-report-issue.rb]: This `printf` is potentially a problem both due to shell interpolation and `printf` interpolation. For the `fill` command, you feed a pre-defined string, but the output you get back (including the user's password) will get fed back to `approve` or `reject`. If the password has a double-quote, a backslash, or a percent, you're going to get bogus results.

You probably want to just pipe straight from ruby to `git credential`, and let the `\n` interpolation happen in the ruby string literal.

Comment (mikemcquaid) [inline cmd/brew-report-issue.rb]: > You probably want to just pipe straight from ruby to git credential, and let the \n interpolation happen in the ruby string literal.

@peff Can you elaborate a little here; not 100% sure what you mean.

Comment (peff) [inline cmd/brew-report-issue.rb]: In the original, the shell `printf` was converting `\n` to newline, but you can do that in ruby, and get rid of `printf` entirely. And that saves you the interpolation headaches. E.g.:

```ruby
# will mess up some strings
def one(cmd, input)
	`printf "#{input}" | #{cmd}`
end

# will feed input verbatim to command
def two(cmd, input)
	IO.popen(cmd, mode="w+") {|io|
		io.print input
		io.close_write
		io.read
	}
end

# just a stand-in for git-credential
cmd = 'sed "s/^/sed: /"'
# this has some tricky sequences in it; the single
# quote means ruby doesn't interpolate \n
msg = 'something with %s and \n in it'

print "one: ", one(cmd, msg), "\n"
print "two: ", two(cmd, msg), "\n"

# but here we use double quotes, and ruby expands \n
print two(cmd, "ruby will interpolate this\nnewline; no need for printf")
```

which should produce:

```
one: sed: something with  and 
sed:  in it
two: sed: something with %s and \n in it
sed: ruby will interpolate this
sed: newline; no need for printf
```

Does that make sense?

I'm not much of a ruby programmer, so my use of popen and general style may not be idiomatic.

Comment (mikemcquaid) [inline cmd/brew-report-issue.rb]: That's great, done that in 4222aa6. Much nicer!

**Comment by peff** [comment]
:+1: Looks good.

**Pull request merged**